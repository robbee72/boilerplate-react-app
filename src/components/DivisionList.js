import React from 'react';
import { Link } from 'react-router-dom';

const TeamListItem = ({ id, homeTown, teamName, conference, division, createdAt }) => (
  <Link className="list-item" to={`/edit/${id}`}>
    <div>
      <h3>{homeTown} {teamName}   {conference} {division}</h3>
  </div>
  </Link>
);

export default TeamListItem;
