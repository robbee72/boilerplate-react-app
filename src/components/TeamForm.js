import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';

export default class TeamForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      homeTown: props.team ? props.team.homeTown : '',
      teamName: props.team ? props.team.teamName : '',
      note: props.team ? props.team.note : '',
      conference: props.team ? props.team.conference : '',
      division: props.team ? props.team.division : '',
      createdAt: props.team ? moment(props.team.createdAt) : moment(),
      calendarFocused: false,
      error: ''
    };
  }
  onHomeTownChange = (e) => {
    const homeTown = e.target.value;
    this.setState(() => ({ homeTown }));
  };
  onTeamNameChange = (e) => {
    const teamName = e.target.value;
    this.setState(() => ({ teamName }));
  };
  onNoteChange = (e) => {
    const note = e.target.value;
    this.setState(() => ({ note }));
  };
  onConferenceChange = (e) => {
    const conference = e.target.value;

    if (!conference || conference.match) {
      this.setState(() => ({ conference }));
    }
  };
  onDivisionChange = (e) => {
    const division = e.target.value;

    if (!division || division.match) {
      this.setState(() => ({ division }));
    }
  };
  onDateChange = (createdAt) => {
    if (createdAt) {
      this.setState(() => ({ createdAt }));
    }
  };
  onFocusChange = ({ focused }) => {
    this.setState(() => ({ calendarFocused: focused }));
  };
  onSubmit = (e) => {
    e.preventDefault();

    if (!this.state.homeTown || !this.state.teamName || !this.state.conference || !this.state.division) {
      this.setState(() => ({ error: 'Please provide Team\'s, Name, conference, and division.' }));
    } else {
      this.setState(() => ({ error: '' }));
      this.props.onSubmit({
        homeTown: this.state.homeTown,
        teamName: this.state.teamName,
        conference: this.state.conference,
        division: this.state.division,
        createdAt: this.state.createdAt.valueOf(),
        note: this.state.note
      });
    }
  };
  render() {
    return (
      <form className="form" onSubmit={this.onSubmit}>
        {this.state.error && <p className="form__error">{this.state.error}</p>}
        <input
          type="text"
          placeholder="Home Town"
          autoFocus
          className="text-input"
          value={this.state.homeTown}
          onChange={this.onHomeTownChange}
        />
        <input
          type="text"
          placeholder="Team Name"
          autoFocus
          className="text-input"
          value={this.state.teamName}
          onChange={this.onTeamNameChange}
        />
        <input
          type="text"
          placeholder="Conference"
          className="text-input"
          value={this.state.conference}
          onChange={this.onConferenceChange}
        />
        <input
          type="text"
          placeholder="Division"
          className="text-input"
          value={this.state.division}
          onChange={this.onDivisionChange}
        />
        <SingleDatePicker
          date={this.state.createdAt}
          onDateChange={this.onDateChange}
          focused={this.state.calendarFocused}
          onFocusChange={this.onFocusChange}
          numberOfMonths={1}
          isOutsideRange={() => false}
        />
        <textarea
          placeholder="Add a note for your team (optional)"
          className="textarea"
          value={this.state.note}
          onChange={this.onNoteChange}
        >
        </textarea>
        <div>
          <button className="button">Save Team</button>
        </div>
      </form>
    )
  }
}
