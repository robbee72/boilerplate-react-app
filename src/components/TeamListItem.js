import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';


const TeamListItem = ({ id, homeTown, teamName, conference, division, createdAt }) => (
  <Link className="list-item" to={`/edit/${id}`}>
    <div>
      <h4>{homeTown} {teamName}   {conference} {division}</h4>

    <span>{moment(createdAt).format('MMMM Do, YYYY')}</span>
  </div>
  </Link>
);

export default TeamListItem;
