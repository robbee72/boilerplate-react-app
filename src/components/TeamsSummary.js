import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import selectTeams from '../selectors/teams';
import selectTeamsTotal from '../selectors/teams-total';

export const TeamsSummary = ({ teamCount, teamsTotal }) => {
  const teamWord = teamCount === 1 ? 'team' : 'teams' ;

  return (
      <div className="page-header">
        <div className="content-container">
          <h3 className="page-header__title">Standing <span>{teamCount}</span> {teamWord}  <span>3</span></h3>
          <div className="page-header__actions">
            <Link className="button" to="/create">Add Team</Link>
          </div>
        </div>
      </div>
    );
  };

const mapStateToProps = (state) => {
  const visibleTeams = selectTeams(state.teams, state.filters);

  return {
    teamCount: visibleTeams.length,
    teamsTotal: selectTeamsTotal(visibleTeams)
  };
};

export default connect(mapStateToProps)(TeamsSummary);
