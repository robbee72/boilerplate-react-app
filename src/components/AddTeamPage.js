import React from 'react';
import { connect } from 'react-redux';
import TeamForm from './TeamForm';
import { startAddTeam } from '../actions/teams';

export class AddTeamPage extends React.Component {
  onSubmit = (team) => {
    this.props.startAddTeam(team);
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add Team</h1>
          </div>
        </div>
        <div className="content-container">
          <TeamForm
            onSubmit={this.onSubmit}
          />
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddTeam: (team) => dispatch(startAddTeam(team))
});

export default connect(undefined, mapDispatchToProps)(AddTeamPage);
