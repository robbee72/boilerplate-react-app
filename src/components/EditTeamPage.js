import React from 'react';
import { connect } from 'react-redux';
import TeamForm from './TeamForm';
import { startEditTeam, startRemoveTeam } from '../actions/teams';

export class EditTeamPage extends React.Component {
  onSubmit = (team) => {
    this.props.startEditTeam(this.props.team.id, team);
    this.props.history.push('/');
  };
  onRemove = () => {
    this.props.startRemoveTeam({ id: this.props.team.id });
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Team</h1>
          </div>
        </div>
        <div className="content-container">
          <TeamForm
            team={this.props.team}
            onSubmit={this.onSubmit}
          />
          <button className="button button--secondary" onClick={this.onRemove}>Remove Team</button>
        </div>
      </div>
    );
  }
};

const mapStateToProps = (state, props) => ({
  team: state.teams.find((team) => team.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
  startEditTeam: (id, team) => dispatch(startEditTeam(id, team)),
  startRemoveTeam: (data) => dispatch(startRemoveTeam(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditTeamPage);
