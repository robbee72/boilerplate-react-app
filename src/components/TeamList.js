import React from 'react';
import { connect } from 'react-redux';
import TeamListItem from './TeamListItem';
import selectTeams from '../selectors/teams';

export const TeamList = (props) => (
  <div className="content-container">
    <div className="list-header">
      <div className="show-for-mobile">Teams</div>
      <div className="show-for-desktop">Team</div>
      <div className="show-for-desktop">Division</div>
    </div>
    {
      props.teams.length === 0 ? (
        <p>No teams</p>
      ) : (
          props.teams.map((team) => {
            return <TeamListItem key={team.id} {...team} />;
          })
        )
    }
  </div>
);

const mapStateToProps = (state) => {
  return {
    teams: selectTeams(state.teams, state.filters)
  };
};

export default connect(mapStateToProps)(TeamList);
