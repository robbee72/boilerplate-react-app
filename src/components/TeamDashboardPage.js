import React from 'react';
import TeamList from './TeamList';
import TeamListFilters from './TeamListFilters';
import TeamsSummary from './TeamsSummary';

const TeamDashboardPage = () => (
  <div>
    <TeamsSummary />
    <TeamListFilters />
    <TeamList />
  </div>
);

export default TeamDashboardPage;
