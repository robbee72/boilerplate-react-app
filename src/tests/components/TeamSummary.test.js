import React from 'react';
import { shallow } from 'enzyme';
import { TeamsSummary } from '../../components/TeamsSummary';

test('should correctly render TeamsSummary with 1 team', () => {
  const wrapper = shallow(<TeamsSummary teamCount={1} teamsTotal={235} />);
  expect(wrapper).toMatchSnapshot();
});

test('should correctly render TeamsSummary with multiple teams', () => {
  const wrapper = shallow(<TeamsSummary teamCount={23} teamsTotal={23512340987} />);
  expect(wrapper).toMatchSnapshot();
});
