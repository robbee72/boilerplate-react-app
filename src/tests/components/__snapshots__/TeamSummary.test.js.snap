// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render TeamsSummary with 1 team 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      team
       totalling 
      <span>
        $2.35
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Team
      </Link>
    </div>
  </div>
</div>
`;

exports[`should correctly render TeamsSummary with multiple teams 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        23
      </span>
       
      teams
       totalling 
      <span>
        $235,123,409.87
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Team
      </Link>
    </div>
  </div>
</div>
`;
